evm/src/fuzz/mod.rs

```diff
@@ -178,6 +178,15 @@ impl<'a> FuzzedExecutor<'a> {
         };
 
         match run_result {
+            // Currently the only operation that can trigger proptest global rejects is the
+            // `vm.assume` cheatcode, thus we surface this info to the user when the fuzz test
+            // aborts due to too many global rejects, making the error message more actionable.
+            Err(TestError::Abort(reason)) if reason.message() == "Too many global rejects" => {
+                result.reason = Some(format!(
+                    "The `vm.assume` cheatcode rejected too many inputs ({} allowed)",
+                    self.runner.config().max_global_rejects
+                ));
```

Comment:
this looks good,

we could start having dedicated error types for all of this.

so I'd suggest, let's start with error.rs and 
```rust
#[derive(thiserror::Error)
#[error("The....")
struct TooManyGlobalRejects();
```

and then `TooManyGlobalRejects(self.runner.config().max_global_rejects).to_string()` here

Reply:
added `error.rs` and `FuzzError` struct
thanks for the suggestion